Adding Nomina Catalogs

AndreyPootMay opened the pull request:
Adding the following nomina catalogs:

| Catálogo               | Método                 |
| ---------------------- | ---------------------- |
| n_TipoNomina           | tiposNominas           |
| n_TipoJornada          | tiposJornadas          |
| n_OrigenesRecursos     | nOrigenesRecursos      |
| n_Bancos               | nBancos                |
| n_PeriodicidadesPagos  | nPeriodicidadesPagos   |
| n_RiesgosPuestos       | nRiesgosPuestos        |
| n_TiposDeducciones     | nTiposDeducciones      |
| n_TiposHoras           | nTiposHoras            |
| n_TiposIncapacidades   | nTiposIncapacidades    |
| n_TiposOtrosPagos      | nTiposOtrosPagos       |
| n_TiposOtrosPagos      | nTiposOtrosPagos       |
| n_TiposOtrosPagos      | nTiposOtrosPagos       |
| n_TiposPercepciones    | nTiposPercepciones     |
| n_TiposRegimenes       | nTiposRegimenes        |

## Test cases

For now I did not create many tests, but later I will create them on this branch. For now the ones I created or modified worked fine according to phpstan:

### tests/UseCase/SatCatalogosTest.php
![image](https://user-images.githubusercontent.com/28898216/111051571-0dc99000-8422-11eb-97b7-200f658e06b2.png)

### tests/Unit/NOMINA
![image](https://user-images.githubusercontent.com/28898216/111051600-449fa600-8422-11eb-83fa-88a22c362f1a.png)

![image](https://user-images.githubusercontent.com/28898216/111051610-54b78580-8422-11eb-9974-83eea91caa9d.png)

AndreyPootMay committed:
Adding Nomina Catalogs

AndreyPootMay committed:
fix: Estado catalog

AndreyPootMay committed:
Corrections

AndreyPootMay committed:
Correcting database seed and  another catalogs.

AndreyPootMay committed:
Correcting repository, db seed and Nomina Catalogs

AndreyPootMay committed:
Correcting tests

eclipxe13 commented:
Creo que sería mucho mejor que los métodos de acceso se llamen `nomina<Algo>` como `nTiposHoras => nominaTiposHoras`.

eclipxe13 commented on docs/Catalogos.md:
```diff
@@ -40,3 +40,18 @@ Catálogos base de CFDI versión 3.3.
 | c_TipoFactor           | tiposFactores          |
 | c_TipoRelacion         | tiposRelaciones        |
 | c_UsoCFDI              | usosCfdi               |
+| n_TipoNomina           | tiposNominas           |
```
Creo que esta debiera ser una nueva tabla

AndreyPootMay replied:
Listo en `docs/Catalogos.md` están separadas y se corrigió el subfijo de cada catálogo conforme al documento de excel que se publica por el SAT.

eclipxe13 commented on src/SatCatalogos.php:
```diff
@@ -26,8 +26,20 @@
  * @method CFDI\TiposRelaciones         tiposRelaciones();
- * @method CFDI\UsosCfdi                usosCfdi();
-*/
+ * @method Cfdi\UsosCfdi                usosCFDI();
```
Se modificó una línea que era correcta, hay que poner el texto como estaba `CFDI\UsosCfdi`

eclipxe13 commented on src/SatCatalogos.php:
```diff
@@ -26,8 +26,20 @@
- * @method CFDI\UsosCfdi                usosCfdi();
-*/
+ * @method Cfdi\UsosCfdi                usosCFDI();
+ * @method Nomina\TiposNominas          tiposNominas();
```
No se está agregando TiposContratos

eclipxe13 commented on src/Repository.php:
```diff
@@ -71,6 +99,20 @@ class Repository
         self::CFDI_TIPOS_COMPROBANTES,
+        self::NOMINA_TIPOS_NOMINAS,
+        self::NOMINA_TIPOS_JORNADAS,
+        self::NOMINA_ESTADOS,
```
No se está agregando NOMIA_ESTADOS, creo que no se debería incluir la constante

eclipxe13 commented on tests/database-seed.sql:
```diff
@@ -343,4 +343,8 @@ INSERT INTO cfdi_tipos_relaciones VALUES('05','Traslados de mercancias facturado
 INSERT INTO cfdi_usos_cfdi VALUES('G01','Adquisición de mercancias',1,1,'2017-01-01','');
 INSERT INTO cfdi_usos_cfdi VALUES('G02','Devoluciones, descuentos o bonificaciones',1,1,'2017-01-01','');
 INSERT INTO cfdi_usos_cfdi VALUES('G03','Gastos en general',1,1,'2017-01-01','');
+INSERT INTO nomina_tipos_nominas VALUES('E', 'Nómina extraordinaria');
```
Agregar muestras y pruebas de todos los demás catálogos que se están agregando

eclipxe13 commented on src/Nomina/TipoHora.php:
```diff
@@ -0,0 +1,15 @@
+
+class TipoHora extends AbstractEntryIdentifiable
+{
+    public function __construct(string $id, string $texto, int $vigenteDesde, int $vigenteHasta)
```
Definición de método innecesaria

eclipxe13 commented on src/Nomina/TipoIncapacidad.php:
```diff
@@ -0,0 +1,15 @@
+
+class TipoIncapacidad extends AbstractEntryIdentifiable
+{
+    public function __construct(string $id, string $texto, int $vigenteDesde, int $vigenteHasta)
```
Definición de método innecesaria

eclipxe13 commented on src/Nomina/TipoOtroPago.php:
```diff
@@ -0,0 +1,15 @@
+
+class TipoOtroPago extends AbstractEntryIdentifiable
+{
+    public function __construct(string $id, string $texto, int $vigenteDesde, int $vigenteHasta)
```
Definición de método innecesaria

eclipxe13 commented on src/Nomina/TipoPercepcion.php:
```diff
@@ -0,0 +1,15 @@
+
+class TipoPercepcion extends AbstractEntryIdentifiable
+{
+    public function __construct(string $id, string $texto, int $vigenteDesde, int $vigenteHasta)
```
Definición de método innecesaria

eclipxe13 commented on src/Nomina/TipoRegimen.php:
```diff
@@ -0,0 +1,15 @@
+
+class TipoRegimen extends AbstractEntryIdentifiable
+{
+    public function __construct(string $id, string $texto, int $vigenteDesde, int $vigenteHasta)
```
Definición de método innecesaria

eclipxe13 commented on src/Nomina/PeriodicidadPago.php:
```diff
@@ -0,0 +1,15 @@
+
+class PeriodicidadPago extends AbstractEntryIdentifiable
+{
+    public function __construct(string $id, string $texto, int $vigenteDesde, int $vigenteHasta)
```
Definición de método innecesaria

eclipxe13 commented on src/Nomina/RiesgoPuesto.php:
```diff
@@ -0,0 +1,15 @@
+
+class RiesgoPuesto extends AbstractEntryIdentifiable
+{
+    public function __construct(string $id, string $texto, int $vigenteDesde, int $vigenteHasta)
```
Definición de método innecesaria

eclipxe13 commented on src/Nomina/TipoContrato.php:
```diff
@@ -0,0 +1,15 @@
+
+class TipoContrato extends AbstractEntryIdentifiable
+{
+    public function __construct(string $id, string $texto, int $vigenteDesde, int $vigenteHasta)
```
Definición de método innecesaria

eclipxe13 commented on src/Nomina/TipoDeduccion.php:
```diff
@@ -0,0 +1,15 @@
+
+class TipoDeduccion extends AbstractEntryIdentifiable
+{
+    public function __construct(string $id, string $texto, int $vigenteDesde, int $vigenteHasta)
```
Definición de método innecesaria

eclipxe13 reviewed (changes requested):
Creo que es importante agregar más tests, tanto que usen la base de datos de pruebas como propiamente del código agregado

AndreyPootMay committed:
feat: Correcting catalogs

Uploading docs, methods, add new tests, new data inside the sql file.

AndreyPootMay committed:
feat: Fixes

Code fixing style using composer.json scripts, database seeds, new test classes for TipoIncapacidad-TiposIncapacidades, updating docs.

AndreyPootMay committed:
deleting phpunit.result.cache

AndreyPootMay committed:
Separing methods

eclipxe13 committed:
code style

eclipxe13 committed:
Refactor SatCatalogs to have hard definitions (avoid magic)

This will allow to have method names different than class names.

eclipxe13 committed:
Add catalog Nomina Estado

eclipxe13 commented:
Anexo el listado correcto de los catálogos que se están agregando.

Dado que los nombres pueden cambiar sin tener que renombrar las clases, propongo los siguientes antes de integrar a la rama principal.

| Catálogo               | Método                | Propuesta |
| ---------------------- | --------------------- | --------------------- |
| c_Banco                | bancos                | |
| c_Estados              | estados               | |
| c_OrigenRecurso        | origenesRecursos      | |
| c_PeriodicidadPago     | periodicidadesPagos   | |
| c_RiesgoPuesto         | riesgosPuestos        | |
| c_TipoContrato         | tiposContratos        | contratos |
| c_TipoDeduccion        | tiposDeducciones      | deducciones |
| c_TipoHoras            | tiposHoras            | horasExtras |
| c_TipoIncapacidad      | tiposIncapacidades    | incapacidades |
| c_TipoJornada          | tiposJornadas         | jornadas |
| c_TipoNomina           | tiposNominas          | nominas |
| c_TipoOtroPago         | tiposOtrosPagos       | otrosTipoPago |
| c_TipoPercepcion       | tiposPercepciones     | percepciones |
| c_TipoRegimen          | tiposRegimenes        | regimenesContratacion |

Por favor, me gustaría saber su opinión.

AndreyPootMay committed:
New methods name conventions

eclipxe13 commented:
Gran trabajo @AndreyPootMay !
